cg3d solver convergence criteria (+ cleaning)

jm-c opened the pull request:
## What changes does this PR introduce?
Add option (+ new parameter) to specify the CG3D target residual in term of vertical-velocity units (`cg3dTargetResWunit >  0`) and to skip the normalization of cg3d RHS. 
This could be useful in light of issue #856

## What is the current behaviour? 
Always normalize cg3d RHS and specify target residual as non-dimensional number.

## What is the new behaviour 
Implement new option in the same way as done for CD2D solver (with `cg2dTargetResWunit` and `cg2dNormaliseRHS`).

## Does this PR introduce a breaking change? 
No

## Other information:
Also remove un-used pieces of code (in `post_cg3d.F`) and header files `SOLVE_FOR_PRESSURE.h` and `SOLVE_FOR_PRESSURE3D.h`

## Suggested addition to `tag-index`
o model/src:
  - Add new param "cg3dTargetResWunit" to skip the normalization of cg3d RHS
    by specifying the CG3D target residual in term of vertical-velocity units ;
  - Remove header files SOLVE_FOR_PRESSURE{3D}.h ; clean-up "post_cg3d.F" ;
  - switch test exp. global_ocean.cs32x15.viscA4 & short_surf_wave to use new
    param "cg3dTargetResWunit".

mjlosch commented on model/src/solve_for_pressure.F:
```diff
@@ -64,25 +60,29 @@ SUBROUTINE SOLVE_FOR_PRESSURE( myTime, myIter, myThid )
 CEOP
 
 #ifdef ALLOW_NONHYDROSTATIC
-        zeroPsNH = .FALSE.
-c       zeroPsNH = use3Dsolver .AND. exactConserv
-c    &                         .AND. select_rStar.EQ.0
-        zeroMeanPnh = .FALSE.
-c       zeroMeanPnh = use3Dsolver .AND. select_rStar.NE.0
-c       oldFreeSurfTerm = use3Dsolver .AND. select_rStar.EQ.0
-c    &                                .AND. .NOT.zeroPsNH
```
Was this a tests or why was this commented out?

mjlosch reviewed (approved):
Looks good. I have a question about the remove unused code. Maybe you can remind us why it is unused?
I find this on my Mac:
```
Y Y Y Y>11<16 16 16 16 16 16 16 16 13 13 13 16 13 13 13 16  .  .  .  .  .  .  .  . pass  global_ocean.cs32x15.viscA4
Y Y Y Y>16<16 16 16 16 16 16 16 16 16 16 16 16 22 22 22 22  .  .  .  .  .  .  .  . pass  short_surf_wave
```
and for master identically
```
Y Y Y Y>11<16 16 16 16 16 16 16 16 13 13 13 16 13 13 13 13  .  .  .  .  .  .  .  . pass  global_ocean.cs32x15.viscA4
Y Y Y Y>16<16 16 16 16 16 16 16 16 16 16 16 16 22 22 22 22  .  .  .  .  .  .  .  . pass  short_surf_wave
```

Not sure if that's expected.

jm-c commented:
@mjlosch Thanks for your review. I will address your comments soon.

jm-c commented:
@mjlosch As I switched 2 test experiments to skip the normalisation of cg3d-RHS, the output changed (as expected in light of #856) but since I updated the 2 ref. output now the differences you see (vs ref output) are rather small.

Regarding this:
> I have a question about the remove unused code. Maybe you can remind us why it is unused?

It was a long time ago that I added `zeroPsNH` and `zeroMeanPnh` at a time when I did not figure out all the details of the free-surface Non-Hydrostatic code, just to be able to easily test things. But these 2 should have been removed when `selectNHfreeSurf` was added as, by that time,  things were sorted out.

Regarding `putPmEinXvector` pieces of code, it's a little bit different. It's also always false, but when I tried to add a very large (un-balanced) PmE, it was saving a couple of cg2d iterations. So it might be (not sure ?) useful to keep it for now as a way to test, but not very useful for practical apllication (unlikely that the global-mean PmE will  be that large) to justify to change this into a run-time switch.

jm-c commented:
From my side, this PR is ready to be merged. @mjlosch do you have any other questions after I replied yesterday ?
Otherwise, and assuming no one want to add things here could merge this PR soon (like tomorrow ?).

jrscott commented:
@jm-c please hold off merging until I update the doc to reflect this change (will be soon)

jm-c commented:
@jrscott Thanks for the updated doc. Will merge this PR soon (unless you have soemthing to add).
